include/geode/geometry/bounding_box.h

```diff
@@ -56,10 +57,18 @@ namespace geode
 
         bool intersects( const BoundingBox< dimension >& bbox ) const;
 
+        bool intersects( const Ray< dimension >& ray ) const;
+
+        double signed_distance( const Point< dimension >& point ) const;
```

Comment:
Add a bit of doc to tell what happens if the point is inside. It is 0, negative?